src/widgets/totalHrsAndGranteeGraph.js

```diff
@@ -14,25 +14,36 @@ function addOrUpdateResponse(traceIndex, res, xValue, valueToAdd) {
 
   const valueToUse = parseFloat(cleanValue, 10) === 0 ? 0 : parseFloat(cleanValue, 10);
 
-  if (res[traceIndex].x.includes(xValue)) {
-    // Update existing Y value.
-    const indexToUpdate = res[traceIndex].x.indexOf(xValue);
-    res[traceIndex].y[indexToUpdate] += valueToUse;
-  } else {
-    // Add X value and Update Y value.
-    res[traceIndex].x.push(xValue);
-    res[traceIndex].y.push(valueToUse);
-  }
+  res.forEach((responseObject, index) => {
+    if (index === traceIndex) {
+      if (responseObject.x.includes(xValue)) {
+        // Update existing Y value.
+        const indexToUpdate = res[traceIndex].x.indexOf(xValue);
+        // eslint-disable-next-line no-param-reassign
+        responseObject.y[indexToUpdate] += valueToUse;
+      } else {
+        // Add X value and Update Y value.
+        responseObject.x.push(xValue);
+        responseObject.y.push(valueToUse);
+      }
+    } else {
```

Comment:
Probably makes sense to add a comment here to explain what happens in this path of the code and why we do it.